src/MessagePack.UnityClient/Assets/Scripts/MessagePack/package.json

```diff
@@ -0,0 +1,11 @@
+{
+    "name": "com.neuecc.messagepack",
+    "displayName": "MessagePack",
+    "version": "2.2.0",
```

Comment:
Is it important that this json file be hard-coded with this version so it's in the git repo, or is it enough that this file show up in the .unitypackage file? If the latter, we could perhaps add a build step to produce or update this file just in time to be packaged so we don't have to worry about keeping this version number updated independently of the rest of the repo.

Reply:
UPM can refer .git directlyy so require to hard-coded version in this repo.

> we could perhaps add a build step to produce or update this file just in time to be packaged

This is ideal, great.
By the way, UPM also allows for tag and branch specifications, so

Set tag -> Commit version update
then the version of tag and UPM may not match.
(For example, tag=v2.2.1, but packagjson version still 2.1.0)
This is not good.

Reply:
> This is ideal, great.

It sounds like you like the idea of a .unitypackage file that contains a package.json file where the version property has been updated by the build.

> UPM can refer .git directlyy so require to hard-coded version in this repo. By the way, UPM also allows for tag and branch specifications, so

But this makes it sound like possibly a moot point, if we want to support unity pointing at arbitrary commits in our repo. Is that something you want to support? Our version number changes with every commit, so I don't think the goal should be to keep the package.json file in sync exactly with that incrementing number by hand. Can we just call the version "2.2" perhaps, so that it matches the branch's overall version but without the 3rd integer precision that it can't keep up with?

Reply:
Sorry, I think I'm miscommunicating a bit.
Would it be clearer if I used a specific example?

This is the UPM's git dependencies document.
https://docs.unity3d.com/Manual/upm-git.html

For example: refer master:
`"com.neuecc.messagepack": https://github.com/neuecc/MessagePack-CSharp.git?path=src/MessagePack.UnityClient/Assets/Scripts/MessagePack`

refer via tag(as release package):
`"com.neuecc.messagepack": https://github.com/neuecc/MessagePack-CSharp.git?path=src/MessagePack.UnityClient/Assets/Scripts/MessagePack#v2.2.1`

If refer via tag `#v2.2.1` but `package.json` file's version is not `2.2.1`, not good.

Reply:
That helps, thanks. If the user refers to a *branch*, then evidently the `package.json`'s version property can be anything. I _suppose_ the same is true if the user refers to a *tag* as well (that strictly speaking, the version can be anything), but it can be confusing to the user, right?

So can we just specify `2.2` as the package.json version, so the user can use `v2.2.1` as the tag and get a package.json file that says `2.2`?

Reply:
When referring to Git, the version of package.json is only shown in the package manager's GUI screen in the description.
So it works even if the two are different.
However, this is not a good situation because it can be misleading.

Reply:
What would you propose as the goal then? That every version we tag is on a commit that sets the package.json file's version? We could write a script that touches that file, commits the change and then tags, so that the version always matches the tag.

Reply:
The ideal goal is that the package.json version of the tagged revision should be equal to the tag.

By the way, it's an optional requirement, so I don't need to do it if CI or Git Hook makes it more complicated.

Reply:
Tracked by #1000